supersim_test.go

```diff
@@ -1301,3 +1302,114 @@ func TestEthSubscribeNewHeads(t *testing.T) {
+	require.NotNil(t, testSuite.Supersim)
+	require.True(t, len(testSuite.Supersim.Orchestrator.L2Chains()) > 0)
+
+	// Verify that we have 3 chains running
+	networkConfig := testSuite.Supersim.NetworkConfig
+	require.Equal(t, 3, len(networkConfig.L2Configs)) // Should have 3 L2 chains
+
+	// Verify that each local chain's dependency set is overridden correctly
+	for _, l2Config := range networkConfig.L2Configs {
+		require.NotNil(t, l2Config.L2Config)
+		depSet := l2Config.L2Config.DependencySet
+
+		if l2Config.ChainID == 903 {
+			require.Equal(t, 0, len(depSet), "Chain 903 should have empty dependency set since it's not included in user-provided set")
+		} else if l2Config.ChainID == 901 {
+			require.Equal(t, 1, len(depSet), "Chain 901 should have exactly 1 dependency (902)")
+			require.Contains(t, depSet, uint64(902), "Chain 901 should include chain 902")
+		} else if l2Config.ChainID == 902 {
+			require.Equal(t, 1, len(depSet), "Chain 902 should have exactly 1 dependency (901)")
+			require.Contains(t, depSet, uint64(901), "Chain 902 should include chain 901")
+		}
+
+		require.NotContains(t, depSet, l2Config.ChainID, "Chain %d should not include itself in dependency set", l2Config.ChainID)
+	}
+}
+
+func TestDependencySetValidation(t *testing.T) {
+	t.Parallel()
+	testSuite := createTestSuite(t, func(cfg *config.CLIConfig) *config.CLIConfig {
+		cfg.L2Count = 3
+		// Only chains 901 and 902 can communicate, 903 is isolated
+		parsedDeps, err := config.ParseDependencySet("[901,902]")
+		require.NoError(t, err)
+		cfg.DependencySet = parsedDeps
+		return cfg
+	})
+
+	testCases := []struct {
+		name          string
+		sourceChainID uint64
+		destChainID   uint64
+		shouldFail    bool
+		expectedError string
+	}{
+		{
+			name:          "Message from isolated chain should fail",
+			sourceChainID: 903, // Chain 903 is isolated
+			destChainID:   901, // Chain 901 only accepts from 902
+			shouldFail:    true,
+			expectedError: "executing message in block (chain 901) may not execute message from chain 903: not in dependency set",
+		},
+		{
+			name:          "Same-chain message should pass",
+			sourceChainID: 901, // Same chain
+			destChainID:   901, // Same chain
+			shouldFail:    false,
+		},
+		{
+			name:          "Message within dependency set should pass",
+			sourceChainID: 902, // Chain 902 is in 901's dependency set
+			destChainID:   901,
+			shouldFail:    false,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			// Get the destination chain's OpSimulator to test validateDependencySet directly
+			destChain := testSuite.Supersim.Orchestrator.L2Chains()[0]
+			for _, chain := range testSuite.Supersim.Orchestrator.L2Chains() {
+				if chain.Config().ChainID == tc.destChainID {
+					destChain = chain
+					break
+				}
+			}
+
+			// Test the dependency set validation logic
+			err := destChain.(*opsimulator.OpSimulator).ValidateDependencySet(tc.sourceChainID)
```

Comment:
instead of calling `ValidateDependencySet` directly, it would be better to test this fully end to end by attempting to send a message to a chain outside the dependency set. You can use these tests as references for how to send a crosschain message: [test for manually relaying a crossdomain message](https://github.com/ethereum-optimism/supersim/blob/88dcdcee5efa72d5def5b35a39f09b70a5e860d6/supersim_test.go#L603), [test for crossdomain eth bridging with autorelay mode on](https://github.com/ethereum-optimism/supersim/blob/88dcdcee5efa72d5def5b35a39f09b70a5e860d6/supersim_test.go#L1005),